UITests/SlideUpViewSpec.swift

```diff
@@ -0,0 +1,103 @@
+import XCTest
+import Quick
+import Nimble
+
+class SlideUpViewSpec: QuickSpec {
+
+    override func spec() {
+
+        var app: XCUIApplication!
+        var iamView: XCUIElement {
+            app.otherElements["IAMView-SlideUp"]
+        }
+
+        func launchAppIfNecessary(args: String) {
+            guard app == nil || !app.launchArguments.joined(separator: " ").contains(args) else {
+                return
+            }
+            self.continueAfterFailure = false
+            app = XCUIApplication()
+            app.launchArguments.append("--uitesting")
+            app.launchArguments.append(args)
+            app.launch()
+        }
+
+        describe("Slide-up campaign view") {
+
+            context("when clicking X button") {
+
+                beforeEach {
+                    launchAppIfNecessary(args: "-campaignType slide-up-redirect")
+                    if !iamView.exists {
+                        app.buttons["login_successful"].tap()
+                        expect(iamView.exists).toEventually(beTrue())
+                    }
+                }
+
+                it("should close the campaign") {
+                    iamView.buttons["exitButton"].tap()
+                    expect(iamView.exists).to(beFalse())
+                }
+
+                it("should have 44pt touch area") {
+                    let exitButtonCenter = iamView.buttons["exitButton"].coordinate(withNormalizedOffset: CGVector(dx: 0.5, dy: 0.5))
+                    let upperLeftCorner = exitButtonCenter.withOffset(CGVector(dx: -22, dy: -22))
+                    upperLeftCorner.tap()
+                    expect(iamView.exists).to(beFalse())
+
+                    app.buttons["login_successful"].tap() // show the message again
+                    expect(iamView.exists).toEventually(beTrue())
+                    let bottomRightCorner = exitButtonCenter.withOffset(CGVector(dx: 21, dy: 21)) // points on max edges are not counted
+                    bottomRightCorner.tap()
+                    expect(iamView.exists).to(beFalse())
+                }
+            }
+
+            context("with redirect action") {
+
+                beforeEach {
+                    launchAppIfNecessary(args: "-campaignType slide-up-redirect")
+                    if !iamView.exists {
+                        app.buttons["login_successful"].tap()
+                        expect(iamView.exists).toEventually(beTrue())
+                    }
+                }
+
+                afterEach {
+                    app.activate() // returns from the browser
+                    usleep(250000) // wait for app switch animation to finish
```

Comment:
Super long delay is a good catch-all in terms of reliability but I personally prefer either disabling animations (or speeding them up at least) just to make UI tests faster and more deterministic

Reply:
But that's just 0.25s 😞 It's for the system's app switch animation - I don't have any control over it. I know it's not a pretty solution but I could't find any other way... Without this the test runner tries to tap a button when it still "moving"
```
Failed to synthesize event: Failed to scroll to visible (by AX action) Button, label: 'custom_test', error: Error kAXErrorCannotComplete performing AXAction 2003 on element AX element pid: 98091, elementOrHash.elementID: 140614030137392.10. (Underlying Error: Error kAXErrorCannotComplete performing AXAction 2003 on element AX element pid: 98091, elementOrHash.elementID: 140614030137392.10)
```

Reply:
I'm removing those tests that involve app transition (opening url). They are too unstable

Reply:
For the system app switching transitions, does the "Reduce Motion" accessibility setting make a difference?

Reply:
I haven't checked that but I have a feeling that it wouldn't help much